gnd/src/main/java/com/google/android/gnd/system/NetworkManager.java

```diff
@@ -41,6 +44,14 @@ public static boolean isNetworkAvailable(Context context) {
     return networkInfo != null && networkInfo.isConnected();
   }
 
+  /**
+   * Returns a Completable that completes immediately on subscribe if network is available, or fails
+   * in error if not.
+   */
+  public static Completable requireActiveNetwork(Context context) {
```

Comment:
Why are we providing the `context` here as a parameter when there's already a method `isNetworkAvailable()` for the same purpose?

Reply:
It's needed because it's used by `FluentCollectionReference`, which isn't injectable (currently instantiated on demand). Instead, it gets the context from the Firebase API. We could create a Provider with Dagger but that might be overkill for now.